Source/Laser/LaserProfiles.H

```diff
@@ -0,0 +1,156 @@
+#ifndef WARPX_LaserProfiles_H_
+#define WARPX_LaserProfiles_H_
+
+#include <AMReX_REAL.H>
+#include <WarpXParser.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Vector.H>
+
+#include <WarpXParser.H>
+
+#include <map>
+#include <string>
+#include <memory>
+#include <functional>
+#include <limits>
+
+namespace WarpXLaserProfiles {
+
+//_________________________________________Laser parameters shared among classes
```

Comment:
Maybe just use a [Doxygen](https://en.wikipedia.org/wiki/Doxygen) comment, which we can parse with tools and include in our manual when needed, of the form:
```suggestion
/** Common laser profile parameters
 *
 * Parameters for each laser profile as shared among all laser profile classes.
 */
```